### PR TITLE
Enable SendsAtomicWithReceive for ASB

### PR DESCRIPTION
**Pull request by SeanFeldman**
**Warning**: I didn't know how to get the `TransportTransactionMode` that normally would be passed into the transport via `PushSettings`. As a workaround, I have exposed `TransportTransactionMode` read-only property on the `ReceiveComponent`, which in turn is used in `RunningEndpointInstance` to exposes that information.

### CODE REVIEW

**Comment by SeanFeldman** [comment]
@andreasohlund is there a better way to achieve a reading of `TransportTransactionMode `?

**Comment by andreasohlund** [comment]
Since we don’t expose the endpoint configuration users could only change
the transaction mode via some config method we provide so technically we
could just add a .MakeSendsAtomicWithFunctionInvoke() method on the
FunctionAwareAsbEndpoint?

**Comment by SeanFeldman** [comment]
SendsAtomicWithReceive is the default.
When is not possible, the transport will automatically lower it to receive
only. I would not want to force customers to use an extra API such as
`.MakeSendsAtomicWithFunctionInvoke()` when they already have to modify
`host.json` _and_ add an extra argument (`MessageReceiver`) to get
transport transaction work. Does that make sense?

>

**Comment by SeanFeldman** [comment]
SendsAtomicWithReceive is the default.
When is not possible, the transport will automatically lower it to receive
only. I would not want to force customers to use an extra API such as
`.MakeSendsAtomicWithFunctionInvoke()` when they already have to modify
`host.json` _and_ add an extra argument (`MessageReceiver`) to get
transport transaction work. Does that make sense @andreasohlund ?

**Comment by andreasohlund** [comment]
In that case we don’t need to modify the core, just look at the value in
host.json?

**Comment by SeanFeldman** [comment]
That's _if_ we can look at `host.json`.

Worth confirming. Another option, is not to even bother checking auto-complete setting in `host.json`. Instead, if `MessageReceiver` is passed in by user in Function (defined and passed into pipeline call), use transactional send. Even if a user forgets to update `host.json`, SendsAtomicWithReceive will still work.

**Comment by SeanFeldman** [comment]
I actually like the latter option more 🙂

**Comment by andreasohlund** [comment]
:+1:

**Comment by SeanFeldman** [comment]
@andreasohlund Core changes reverted. Now it works as described in https://github.com/ParticularLabs/AzureWorks/pull/1#issuecomment-495931955.

File: NServiceBus.Core/Unicast/RunningEndpointInstance.cs [hunk 1]
```diff
@@ -102,7 +102,7 @@ public Task Unsubscribe(Type eventType, UnsubscribeOptions options)
         }
 
         IBuilder builder;
-        ReceiveComponent receiveComponent;
+        internal ReceiveComponent receiveComponent;
```
Comment (andreasohlund) [inline NServiceBus.Core/Unicast/RunningEndpointInstance.cs]: This can be reverted as well

Comment (SeanFeldman) [reply NServiceBus.Core/Unicast/RunningEndpointInstance.cs]: Ah, missed that. Thank you.

**Review by andreasohlund — APPROVED** [review]

**Pull request merged**